File: Server/src/main/java/jk_5/nailed/server/command/CommandRandomSpawnpoint.java
package jk_5.nailed.server.command;

import net.minecraft.command.*;
import net.minecraft.entity.player.*;

import jk_5.nailed.api.*;
import jk_5.nailed.api.map.*;
import jk_5.nailed.api.map.teleport.*;
import jk_5.nailed.api.player.*;

/**
 * No description given
 *
 * @author jk-5
 */
public class CommandRandomSpawnpoint extends NailedCommand {

    public CommandRandomSpawnpoint() {
        super("randomspawnpoint");
    }

    @Override
    public void processCommandPlayer(Player sender, Map map, String[] args) {
        if(args.length == 0){
            TeleportOptions options = new TeleportOptions();
            options.setLocation(map.getRandomSpawnpoint());
            NailedAPI.getTeleporter().teleportEntity(sender.getEntity(), options);
        }else{
            this.processCommandWithMap(sender.getEntity(), map, args);
        }
    }

    @Override
    public void processCommandWithMap(ICommandSender sender, Map map, String[] args) {
        if(args.length == 1){
            EntityPlayerMP[] players = getPlayersList(sender, args[0]);
            for(EntityPlayerMP player : players){
                TeleportOptions options = new TeleportOptions();
                options.setLocation(map.getRandomSpawnpoint());
                NailedAPI.getTeleporter().teleportEntity(player, options);
            }
        }else{
            throw new WrongUsageException(this.getCommandUsage(sender));
        }
    }
}

File: Server/src/main/scala/jk_5/nailed/server/command/CommandRandomSpawnpoint.scala
package jk_5.nailed.server.command

import net.minecraft.command.ICommandSender
import jk_5.nailed.api.map.Map
import jk_5.nailed.api.player.Player
import jk_5.nailed.api.map.teleport.TeleportOptions
import jk_5.nailed.api.NailedAPI

/**
 * No description given
 *
 * @author jk-5
 */
object CommandRandomSpawnpoint extends ScalaCommand {

  val name = "randomspawnpoint"
  val usage = "/randomspawnpoint [player] - Teleports you or the specified player to a random spawnpoint"

  override def processCommandPlayer(sender: Player, map: Map, args: Array[String]) =
    if(args.length == 0){
      val options = new TeleportOptions()
      options.setLocation(map.getRandomSpawnpoint)
      NailedAPI.getTeleporter.teleportEntity(sender.getEntity, options)
    }else{
      this.processCommandWithMap(sender.getEntity, map, args)
    }

  override def processCommandWithMap(sender: ICommandSender, map: Map, args: Array[String]) =
    if(args.length == 1){
      val players = getPlayersList(sender, args(0))
      for(player <- players){
        val options = new TeleportOptions
        options.setLocation(map.getRandomSpawnpoint)
        NailedAPI.getTeleporter.teleportEntity(player, options)
      }
    }else throw new WrongUsageException("Usage: /randomspawnpoint [player]")
}
